run/radius2john.py

```diff
@@ -1,68 +1,139 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 
-# This software is Copyright (c) 2019 Maxime GOYETTE <maxgoyette0-at-gmail.com>,
+# This software is Copyright (c) 2024, k4amos <k4amos at proton.me>
 # and it is hereby released to the general public under the following terms:
+#
 # Redistribution and use in source and binary forms, with or without
 # modification, are permitted.
-#
+
+# This script is essentially a python version of radius2john.pl written by Didier ARENZANA. 
+# The previous version of radius2john.py was written by Maxime GOYETTE <maxgoyette0-at-gmail.com>
+
+# ---
+
 # Utility to bruteforce RADIUS shared-secret
 # Usage: ./radius2john.py <pcap files>
 #
 # This script depends on Scapy (https://scapy.net)
 # To install: pip install --user scapy
 
-try:
-    from scapy.all import *
-except ImportError:
-    print("scapy is missing, run 'pip install --user scapy' to install it!")
-    exit(1)
+# ---
+
+# Application of two  methods described in http://www.untruth.org/~josh/security/radius/radius-auth.html :
+# "3.3 User-Password Attribute Based Shared Secret Attack"
+# "3.1 Response Authenticator Based Shared Secret Attack"
+
+# For attack 3.3 :
+# we try authentications using a known password, and sniff the radius packets to a pcpap file.
+# This script reads access-request in the pcap file, and dumps the md5(RA+secret) and RA, in a john-friendly format.
+# The password must be always the same, be less then 16 bytes long, and entered in the $PASSWORD variable below.
```

Comment:
As far as I understood, users need to touch the code for attack 3.3 in order to set the `PASSWORD` and `LOGINS` variables. I think it would be much nicer to have commandline options for that. The [argparse](https://docs.python.org/3/library/argparse.html) module comes in handy here. I can help with this if needed.